overlay/utils.sh

```diff
@@ -114,6 +115,60 @@ dumpDepInfo() {
     done
 }
 
+install_crate2() {
+    mkdir -p "$out/lib"
+    # shellcheck disable=SC2046
+    cp $(jq -rR 'fromjson?
+        | select(.reason == "compiler-artifact")
+        | .filenames
+        | map(select(test("\\.(a|rlib|so|dylib)$")))
+        | .[]' < .cargo-build-output) "$out/lib" || :
+
+    mkdir -p "$out/bin"
+    # shellcheck disable=SC2046
+    cp $(jq -rR 'fromjson?
```

Comment:
The errors actually originate from this line if there are no binaries to copy. We can safely silent them by redirecting stderr to /dev/null

Reply:
We pre-emptively create a `bin` and `lib` directory regardless of whether `bin` or `lib` targets actually exist. I don't think we should be doing that, no?

Reply:
Or are these empty directories supposed to be present regardless, by Nixpkgs convention?

Reply:
@trha Also, what about the `grep` errors coming from `$out_dir` being undefined? Can we not guard certain parts of `install_crate2` by whether `out_dir` is set or not?

Reply:
We can check if they're empty and remove them later. It's more straightforward that way.

Reply:
But `lib` isn't ever empty because `.dep-keys` is created there beforehand. And we'd have to go in and ignore errors for many different pieces of the function. I think it would actually be much easier to gate the creation of these directories based on whether `$out_dir` is set or not.

Reply:
Okay, I think I get it now, after some review. I think we can do both approaches. I believe that the two blocks you highlighted can be safely have `2> /dev/null` with no consequence. Likewise, the `grep` and `installed_out_dir` sections can be gated by whether `$out_dir` was set successfully or not.

Reply:
Done in a095422 and 0e258d2.